code/modules/vehicles/multitile/hardpoints.dm

```diff
@@ -31,6 +32,69 @@ Currently only has the tank hardpoints
 	var/max_clips = 1 //1 so they can reload their backups and actually reload once
 	var/buyable = TRUE
 
+/obj/item/hardpoint/Initialize()
+	. = ..()
+	if(starter_ammo)
+		ammo = new starter_ammo
+
+/obj/item/hardpoint/examine(mob/user)
+	. = ..()
+	var/status = health <= 0.1 ? "broken" : "functional"
+	var/span_class = health <= 0.1 ? "<span class = 'danger'>" : "<span class = 'notice'>"
+	if((user?.mind?.cm_skills?.engineer && user.mind.cm_skills.engineer >= SKILL_ENGINEER_METAL) || isobserver(user))
+		switch(PERCENT(health / maxhealth))
+			if(0.1 to 33)
+				status = "heavily damaged"
+				span_class = "<span class = 'warning'>"
+			if(33.1 to 66)
+				status = "damaged"
+				span_class = "<span class = 'warning'>"
+			if(66.1 to 90)
+				status = "slighty damaged"
+			if(90.1 to 100)
+				status = "intact"
+	to_chat(user, "[span_class]It's [status].</span>")
+
+/obj/item/hardpoint/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/ammo_magazine/tank))
+		try_add_clip(W, user)
+		return
+	if(!iswelder(W) && !iswrench(W))
+		return ..()
+	var/repair_delays = 6
+	var/obj/item/tool/repair_tool = /obj/item/tool/weldingtool
+	switch(slot)
+		if(HDPT_PRIMARY)
+			repair_delays = 5
+		if(HDPT_SECDGUN)
+			repair_tool = /obj/item/tool/wrench
+			repair_delays = 3
+		if(HDPT_SUPPORT)
+			repair_tool = /obj/item/tool/wrench
+			repair_delays = 2
+		if(HDPT_ARMOR)
+			repair_delays = 10
+	if(!istype(W, repair_tool))
+		to_chat(user, "<span class='warning'>That's the wrong tool. Use a [iswrench(repair_tool) ? "wrench" : "welder"].</span>")
+		return
+	if(iswelder(W))
```

Comment:
Since you're doing 2 typechecks and until do_after is reworked, at least make it a variable to save on processing.